walker.go

```diff
@@ -167,6 +164,13 @@ func Walk(ctx context.Context, p string, opt *WalkOpt, fn filepath.WalkFunc) err
 			}
 		}
 
+		if walkErr != nil {
+			if skip && errors.Is(walkErr, os.ErrPermission) {
```

Comment:
with this `skip` check in here is it possible to ever reach this code when #121 is applied?

Reply:
Yes, the optimization in #121 only applies in specific cases when only simple patterns are used and we can easily prove that a pattern won't match anything inside a dir. We will still set `skip = true` when we can't prove this and need to test the files inside the dir. We would reach this code when we recurse into a directory in these cases.